File: majorProject/src/main/java/com/aocc/majorproject/ui/UiSelectionHighlight.java
package com.aocc.majorproject.ui;

import android.graphics.Color;

import com.aocc.framework.Graphics;

/** Gamepad / keyboard focus ring — outline only, does not cover button art. */
public final class UiSelectionHighlight {

    public static final int COLOR = Color.rgb(255, 220, 80);
    public static final float STROKE_WIDTH = 3f;

    private UiSelectionHighlight() {
    }

    public static void paintRect(Graphics g, UiBounds bounds) {
        g.drawRectOutline(bounds.x, bounds.y, bounds.width, bounds.height, COLOR, STROKE_WIDTH);
    }

    public static void paintCircle(Graphics g, int centerX, int centerY, int radius) {
        g.drawCircleOutline(centerX, centerY, radius, COLOR, STROKE_WIDTH);
    }
}

File: majorProject/src/main/kotlin/com/aocc/majorproject/ui/UiSelectionHighlight.kt
package com.aocc.majorproject.ui

import android.graphics.Color
import com.aocc.framework.Graphics

/** Gamepad / keyboard focus ring — outline only, does not cover button art. */
object UiSelectionHighlight {

    @JvmField
    val COLOR: Int = Color.rgb(255, 220, 80)

    @JvmField
    val STROKE_WIDTH: Float = 3f

    @JvmStatic
    fun paintRect(g: Graphics, bounds: UiBounds) {
        g.drawRectOutline(bounds.x, bounds.y, bounds.width, bounds.height, COLOR, STROKE_WIDTH)
    }

    @JvmStatic
    fun paintCircle(g: Graphics, centerX: Int, centerY: Int, radius: Int) {
        g.drawCircleOutline(
            centerX.toFloat(),
            centerY.toFloat(),
            radius.toFloat(),
            COLOR,
            STROKE_WIDTH
        )
    }
}
